services/checks/checks.go

```diff
@@ -211,6 +213,23 @@ func (s *Service) StartChecks(ctx context.Context) error {
 
 	s.collectChecks(nCtx)
 	s.executeChecks(nCtx)
+
+	go func() {
+		t := time.NewTicker(s.alertsRegistry.resendInterval)
```

Comment:
Shouldnt it be somewhere where we send alerts? Here or in alertmanagerService?

Reply:
actually we wanted to move the responsibility of calling `alertmanager.SendAlerts` to its caller (see [this](https://github.com/percona/pmm-managed/pull/416/files#diff-a8171d88a4e24ab2acd4b471b52230faR133-R135)) and since we want to keep resending alerts at an interval I moved the ticker here

The last 2 points in the ticket's [description](https://jira.percona.com/browse/SAAS-236) also mention that SendAlerts call to be moved to StartChecks method and called in a separate goroutine

Reply:
That code is not correct. `StartChecks` is being called from the API. Two calls means that we create two goroutines that never finish. 1000 calls – 1000 goroutines. But we should have one goroutine for that.

Last two points in description mean that this goroutine should be in a separate method, and there we should be able trigger another send loop manually, without waiting 24 hours.

Reply:
> Last two points in description mean that this goroutine should be in a separate method, and there we should be able trigger another send loop manually, without waiting 24 hours.

I have moved the goroutine to a separate method, but one thing I can't understand is "trigger another send loop manually" which method will call the separate method?

Reply:
First of all, method `resendAlerts` currently does nothing :)

Reply:
have updated the code, `resendAlerts` is now called from `Run` after starting the checks so one resend loop will always be running now what I don't understand is what you mean by manually triggering the loop, we don't want the loop in StartChecks so I am assuming that we don't want the user to be manually triggering the loop so which part of our code will trigger it and why?

Reply:
> First of all, method resendAlerts currently does nothing :)

That's literally it:

```go
// resendAlerts can be used for manually triggerring a SendAlerts loop
func (s *Service) resendAlerts(ctx context.Context) {
	go func() {
		t := time.NewTicker(s.alertsRegistry.resendInterval)
		defer t.Stop()

		for {
			select {
			case <-ctx.Done():
				return
			case <-t.C:
				// nothing, continue for loop
			}
		}
	}()
}
```

All that code does is creates a goroutine that does nothing.

Once it starts doing _something_ (like actually sending alerts), I think other things will fall into places.

Reply:
not sure how that happened, I remember having a call to SendAlerts there and pushing the code  :confused:

anyway I have added it and pushed it now, sorry for bothering you without properly updating the code @AlekSi